src/indicators_1d.jl

```diff
@@ -0,0 +1,225 @@
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+#! format: noindent
+
+# this method is used when the indicator is constructed as for shock-capturing volume integrals
+# empty cache is default
+function create_cache(::Type{<:IndicatorNeuralNetwork},
+                      equations::AbstractEquations{1}, basis::LobattoLegendreBasis)
+    return NamedTuple()
+end
+
+# cache for NeuralNetworkPerssonPeraire-type indicator
+function create_cache(::Type{IndicatorNeuralNetwork{NeuralNetworkPerssonPeraire}},
+                      equations::AbstractEquations{1}, basis::LobattoLegendreBasis)
+    alpha = Vector{real(basis)}()
+    alpha_tmp = similar(alpha)
+    A = Array{real(basis), ndims(equations)}
+
+    prototype = A(undef, nnodes(basis))
+    indicator_threaded = [similar(prototype) for _ in 1:Threads.nthreads()]
+    modal_threaded = [similar(prototype) for _ in 1:Threads.nthreads()]
+
+    return (; alpha, alpha_tmp, indicator_threaded, modal_threaded)
+end
+
+# cache for NeuralNetworkRayHesthaven-type indicator
+function create_cache(::Type{IndicatorNeuralNetwork{NeuralNetworkRayHesthaven}},
+                      equations::AbstractEquations{1}, basis::LobattoLegendreBasis)
+    alpha = Vector{real(basis)}()
+    alpha_tmp = similar(alpha)
+    A = Array{real(basis), ndims(equations)}
+
+    prototype = A(undef, nnodes(basis))
+    indicator_threaded = [similar(prototype) for _ in 1:Threads.nthreads()]
+    neighbor_ids = Vector{Int}(undef, 2)
+
+    return (; alpha, alpha_tmp, indicator_threaded, neighbor_ids)
+end
+
+# this method is used when the indicator is constructed as for AMR
+function create_cache(typ::Type{<:IndicatorNeuralNetwork},
+                      mesh, equations::AbstractEquations{1}, dg::DGSEM, cache)
+    create_cache(typ, equations, dg.basis)
+end
+
+function (indicator_ann::IndicatorNeuralNetwork{NeuralNetworkPerssonPeraire})(u::AbstractArray{
+                                                                                               <:Any,
+                                                                                               3
+                                                                                               },
+                                                                              mesh,
+                                                                              equations,
+                                                                              dg::DGSEM,
+                                                                              cache;
+                                                                              kwargs...)
+    @unpack indicator_type, alpha_max, alpha_min, alpha_smooth, alpha_continuous, alpha_amr, variable, network = indicator_ann
+
+    @unpack alpha, alpha_tmp, indicator_threaded, modal_threaded = indicator_ann.cache
```

Comment:
I guess you need SimpleUnPack.jl for this?

Reply:
Because of
https://github.com/trixi-framework/Trixi.jl/blob/158e53372cd168e3c2b243f70e3b9faa62c700c3/src/Trixi.jl#L71
I don't _need_ it explicitly. You think it is necessary to add for this repo?

Reply:
I prefer specifying dependencies directly - but this is not a high-priority repo, so we can keep it as it is